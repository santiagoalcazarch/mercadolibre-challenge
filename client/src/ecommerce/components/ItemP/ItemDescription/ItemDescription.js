
import React from 'react'
import PropTypes from 'prop-types'

/**
 * Muestra la descripción de un item en particular
 */
const ItemDescription = ({ itemDescription }) => {
  return (
    <div className="item-l-desc-content mt-4">
      <div>
        <h3> Descripción del producto </h3>
      </div>
      <p className="item-l-description">
        { itemDescription ?? "" }
      </p>
    </div>
  )
}

ItemDescription.propTypes = {
  itemDescription: PropTypes.string
}

export default ItemDescription