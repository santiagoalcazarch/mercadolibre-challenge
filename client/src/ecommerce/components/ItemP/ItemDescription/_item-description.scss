
.item-l-desc-content{

  h3{
    font-size: 28px;
  }
  
  .item-l-description{
    white-space: pre-line;
    font-size: 16px;
  }
}
