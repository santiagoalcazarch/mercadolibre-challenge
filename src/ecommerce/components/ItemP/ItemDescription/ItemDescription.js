
import React from 'react'
import PropTypes from 'prop-types'

const ItemDescription = props => {
  return (
    <div className="item-l-desc-content mt-4">
      <div>
        <h3> Descripci√≥n del producto </h3>
      </div>
      <p className="item-l-description">
        lorem
      </p>
    </div>
  )
}

ItemDescription.propTypes = {}

export default ItemDescription