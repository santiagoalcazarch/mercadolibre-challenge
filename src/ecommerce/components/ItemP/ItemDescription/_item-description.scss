
.item-l-desc-content{

  h3{
    font-size: 28px;
  }
  
  .item-l-description{
    font-size: 16px;
  }
}
