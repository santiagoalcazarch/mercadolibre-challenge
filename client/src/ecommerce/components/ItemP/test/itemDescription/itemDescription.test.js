
import { shallow } from "enzyme";
import ItemDescription from "../../ItemDescription/ItemDescription";

describe('ItemDescription component test', () => {

  const itemDescription = "lorem ipsum dolor sit amet, consectetur adipiscing";

  const wrapper = shallow( <ItemDescription itemDescription={itemDescription} /> );

  test('Debería renderizar el componente correctamente', () => { 
    expect(wrapper).toMatchSnapshot();
  });

  test('Debería renderizar el parrafo de la descripción enviada por parametro', () => { 
    const recivedText = wrapper.find("p").text();
    expect(recivedText).toBe(itemDescription);
  });

  test('Debería renderizar como vacío le parrafo de descripción enviada por parametro', () => { 
    const wrapper = shallow( <ItemDescription itemDescription={undefined} /> );

    const recivedText = wrapper.find("p").text();
    expect(recivedText).toBe("");
  });

})