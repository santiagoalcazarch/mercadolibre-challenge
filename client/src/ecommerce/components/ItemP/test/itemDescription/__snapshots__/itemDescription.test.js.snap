// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemDescription component test Debería renderizar el componente correctamente 1`] = `
<div
  className="item-l-desc-content mt-4"
>
  <div>
    <h3>
       Descripción del producto 
    </h3>
  </div>
  <p
    className="item-l-description"
  >
    lorem ipsum dolor sit amet, consectetur adipiscing
  </p>
</div>
`;
